poutyne/framework/metrics/epoch_metrics/fscores.py

```diff
@@ -0,0 +1,259 @@
+"""
```

Comment:
I'm really not sure of the plus value to remove dependancy from AllenNLP vs having a bigger code base to maintain.

> If we keep this code we should at least copy too the tests.

Reply:
The plus value is that, right now, we cannot add AllenNLP as a dependency since it is not compatible yet with PyTorch 1.2. So the people using Poutyne wanting to use the F1 score would be obligated to use PyTorch 1.1.

Reply:
Ok so the plan is to add it, and leave it after ?

Reply:
I don't think having AllenNLP as a dependency is a good a idea. It comes itself with a lot of dependencies I would like to avoid. I think the project would be too linked with AllenNLP and we would be obligated to follow their development as well as PyTorch's. Also, the code for FBeta is not that huge and this will allow us to customize it more.

Reply:
I just copied the tests from AllenNLP.

Reply:
Agree with @freud14 here: Poutyne is basically a wrapper around Pytorch and should therefore only be strongly linked to Pytorch as a library. Any other big library dependence like the one we introduced with AllenNLP is more annoying than simply copy-pasting the code for further modifications if needed. The more libraries you strongly connect to, the more your library has to be constantly updated when those connected libraries are also updated.

Reply:
Fair enough for the argument of preferring binding with torch and be minimal. 
I still think we open for copy-paste code base but for now it's good enough.